### PR TITLE
feat: first implementation

### PR DESCRIPTION
**Pull request by atian25**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: app/extend/application.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+const is = require('is-type-of');
+const routerMap = {};
+
+const methods = [
```
Comment (dead-horse) [inline app/extend/application.js]: methods 可以直接用 require('methods') 吧？

File: app/extend/application.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+module.exports = {
+  /**
+   * get sub router
+   *
+   * @method Application#getRouter
+   * @param {String} prefix - sub router prefix
+   * @param {...Function} [middlewares] - optional middlewares
+   * @return {Router} Return sub router with special prefix
+   */
+  getRouter(prefix, ...middlewares) {
```
Comment (dead-horse) [inline app/extend/application.js]: namespace ?

Comment (dead-horse) [reply app/extend/application.js]: `app.router.namespace` 或者 `app.subRouter` ?

File: app/extend/application.js (hunk 1 above)
Comment (atian25) [reply app/extend/application.js]: https://github.com/jshttp/methods/blob/master/index.js#L40-L69

好像那个库会输出很多乱七八糟的方法，我看 egg-core 也是自己写死几个的。

File: app/extend/application.js (hunk 2 above)
Comment (atian25) [reply app/extend/application.js]: `app.router.namespace` +1

**Pull request merged**